### ISSUE TITLE
ERROR - Exports must be a statement at the top-level of a module when compiling es6 output from typescript compiler

### ISSUE THREAD

**Issue by davityle** [google/closure-compiler#2429]
Typescript output for an enum fails to compile with closure compiler because it assigns an export inside of a statement. For example

```typescript
export enum Foo {}
```

transpiles to

```javascript
var Foo;
(function (Foo) {
})(Foo = exports.Foo || (exports.Foo = {}));
```

as demonstrated [here](https://www.typescriptlang.org/play/#src=export%20enum%20Foo%20%7B%7D)

When closure compiler is run on that code it results in the error `ERROR - Exports must be a statement at the top-level of a module`

I don't believe that this should cause an error. Es6 supports this syntax. At the very least a there should be an option to make this into a warning or `@suppress` it.

**Comment by MatrixFrog** [comment]
cc @alexeagle

**Comment by alexeagle** [comment]
Did you build your TypeScript code with `tsickle`? https://www.npmjs.com/package/tsickle

**Comment by davityle** [comment]
@alexeagle this is output from Angular 2's `ngc`

**Comment by davityle** [comment]
with `"googleClosureOutput": true`

**Comment by alexeagle** [comment]
I don't think that's the right option, should be
 "angularCompilerOptions": {
        "annotateForClosureCompiler": true
    }

**Comment by davityle** [comment]
`"googleClosureOutput"` is something that we added at lucid last summer. updating it to "annotateForClosureCompiler" fixed it. (amazingly it compiled and worked with only a single runtime error before updating closure where this error appeared)

**Comment by alexeagle** [comment]
Oh, hi, you didn't use the magic word. I remember that option now - it was from the early prototype of tsc_wrapped that you guys forked. Glad it's fixed.
